src/internal_rules.jl

```diff
@@ -815,3 +815,25 @@ function EnzymeRules.forward(func::Const{typeof(ldiv!)},
         end
     end
 end
+
+# Ranges
+# Float64 ranges in Julia use bitwise `&` with higher precision
+# to correct for numerical error, thus we put rules over the
+# operations as this is not directly differentiable
+
+getval(x) = hasproperty(x, :val) ? x.val : x
+function EnzymeRules.forward(func::Const{Colon}, RT::Type{<:Union{Const, DuplicatedNoNeed, Duplicated}}, start, step, stop)
+    ret = func.val(getval.((start, step, stop))...)
```

Comment:
Since it's three things can you just do .val for all of them [there isn't a question that they would have the property all Enzyme.Annotation values have .val\